src/mlmc/sample_storage_hdf.py

```diff
@@ -151,6 +156,35 @@ def n_finished(self):
 
         return n_finished
 
+    def unfinished_ids(self):
+        """
+        List of unfinished ids
+        :return: list
+        """
+        unfinished = []
+
+        for level in self._level_groups:
```

Comment:
Seems that `SampleStorageHDF` is aware of the HDF structure, so I ask where is the border between SampleSTORAGEHDF and the hdf5? There are historic reasons, but could we simplify things putting it into a single class?